pkg/deploy/controller/deploymentconfig/factory.go

```diff
@@ -51,6 +52,10 @@ func NewDeploymentConfigController(
 
 		rcLister:       rcInformer.Lister(),
 		rcListerSynced: rcInformer.Informer().HasSynced,
+		rcControl: oscontroller.RealRCControl{
+			KubeClient: internalKubeClientset,
```

Comment:
i would create `NewRealRCControl()` and pass the client and recorded and make them unexported unless.